src/imitation/testing/expert_trajectories.py

```diff
@@ -0,0 +1,167 @@
+) -> Sequence[types.TrajectoryWithRew]:
+    """Generate or load expert trajectories from cache.
+
+    Args:
+        cache_path: A path to the folder to be used as cache for the expert
+            trajectories.
+        env_id: The environment to generate trajectories for.
+        num_trajectories: The number of trajectories to generate.
+
+    Returns:
+        A list of trajectories with rewards.
+    """
+    environment_cache_path = pathlib.Path(cache_path) / hfsb3.EnvironmentName(env_id)
+    environment_cache_path.mkdir(parents=True, exist_ok=True)
+
+    trajectories_path = environment_cache_path / "rollout.npz"
+
+    with FileLock(environment_cache_path / "rollout.npz.lock"):
+        try:
+            trajectories = types.load_with_rewards(trajectories_path)
+        except (FileNotFoundError, pickle.PickleError):  # pragma: no cover
+            warnings.warn(
+                "Recomputing expert trajectories due to the following error when "
+                "trying to load them:\n" + traceback.format_exc(),
+            )
+            trajectories = generate_expert_trajectories(env_id, num_trajectories)
+            types.save(trajectories_path, trajectories)
+
+    if len(trajectories) >= num_trajectories:
+        return trajectories[:num_trajectories]
+    else:
+        # If it is not enough, just throw clear the cache and generate more.
+        trajectories_path.unlink()
+        return lazy_generate_expert_trajectories(cache_path, env_id, num_trajectories)
+
+
+def make_expert_transition_loader(
+    cache_dir: Path,
+    batch_size,
+    expert_data_type,
+    env_name: str,
+    num_trajectories: int = 1,
+):
+    """Creates different kinds of pytorch data loaders for expert transitions.
+
+    Args:
+        cache_dir: The directory to use for caching the expert trajectories.
+        batch_size: The batch size to use for the data loader.
+        expert_data_type: The type of expert data to use. Can be one of "data_loader",
+            "ducktyped_data_loader", "transitions".
+        env_name: The environment to generate trajectories for.
+        num_trajectories: The number of trajectories to generate.
+
+    Raises:
+        ValueError: If `expert_data_type` is not one of the supported types.
+
+    Returns:
+        A pytorch data loader for expert transitions.
+    """
+    trajectories = lazy_generate_expert_trajectories(
+        cache_dir,
+        env_name,
+        num_trajectories,
+    )
+    transitions = rollout.flatten_trajectories(trajectories)
+
+    if len(transitions) < batch_size:
+        # If we have less transitions than the batch size, we estimate the trajectory
+        # length and generate enough trajectories to fill the batch size.
+        trajectory_length = len(transitions) // len(trajectories)
+        min_required_trajectories = math.ceil(batch_size / trajectory_length)
+        transitions = rollout.flatten_trajectories(
+            lazy_generate_expert_trajectories(
```

Comment:
This might return data that overlaps with `trajectories` -- is that OK? (Probably?)

Reply:
Yeah, I think this is probably fine as-is, the solutions I can think of would add more complexity than they're worth.